rest/data/test.go

```diff
@@ -58,6 +58,38 @@ func (tc *DBTestConnector) FindTestsByTaskId(taskId, testId, testName, status st
 	return res, nil
 }
 
+func (tc *DBTestConnector) FindTestsByTaskIdFilterSortPaginate(taskId, filter, sortBy string, sortDir, page, limit, execution int) ([]testresult.TestResult, error) {
+	t, err := task.FindOneId(taskId)
+	if err != nil {
+		return []testresult.TestResult{}, gimlet.ErrorResponse{
```

Comment:
nit: I would return nil rather than an empty array for these error cases

Reply:
I was wondering your reasoning for that. I thought it would easier for the UI to consume an empty list than null since it is also another way to represent that no results were found without switching types.

Reply:
That's fine, but you're returning an empty array in the resolver regardless. Returning nil here would make a bit more sense from a go perspective, but it's not a big deal